Add SSH troubleshooting guide (#190)
mit-orcd/orcd-docs-edit: docs/accessing-orcd/ssh-troubleshooting.md

---
tags:
 - SSH
 - Troubleshooting
 - Logging In
---

# SSH Troubleshooting

## Connection Refused or Timeout

Make sure you are using the correct login command:
```bash
ssh USERNAME@orcd-login.mit.edu
```

> If the connection hangs or times out, you may be on a network that blocks SSH. Try a different network or contact [orcd-help@mit.edu](mailto:orcd-help@mit.edu).

Reviewer: Maybe add a link in this section to the "IP Address Blocked" section below?

```suggestion
If the connection hangs or times out, you may be on a network that blocks SSH, or your IP address has been blocked (see [below](#ip-address-blocked)). Try a different network or contact [orcd-help@mit.edu](mailto:orcd-help@mit.edu).
```

## Permission Denied

**Check your username:** Make sure you are using your MIT Kerberos username.

**Check your SSH key permissions:** Run these commands on your laptop:
```bash
chmod 700 ~/.ssh
chmod 600 ~/.ssh/id_rsa
```

**Key not uploaded:** If you have not yet set up SSH keys, see the [SSH Key Setup](ssh-setup.md) page.

!!! note
    Even with SSH keys set up, you will still be prompted for your MIT Kerberos password and Duo. This is expected.

## Host Key Warning

If you see a message like `WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED` or `This host key is known by the following other names`, run:
```bash
ssh-keygen -R orcd-login.mit.edu
```

> Then reconnect and type `yes` when prompted. This is common after system maintenance.

Reviewer: ```suggestion
Then reconnect and type `yes` when prompted. This is sometimes happens after system maintenance.
```

If the block does not clear after 10 minutes, open a ticket with [orcd-help@mit.edu](mailto:orcd-help@mit.edu) and include your IP address. You can find it with:

=== "macOS/Linux"
    ```bash
    curl ifconfig.me
    ```
=== "Windows"
    ```
    curl ifconfig.me
    ```
    If that does not work, try:
    ```
    (Invoke-WebRequest -Uri "ifconfig.me").Content
    ```

## Account Locked

10 failed Duo attempts will lock your account. This lock is managed by [MIT IS&T](https://ist.mit.edu/duo-security/duo), not ORCD, and will lift automatically after 90 minutes provided no further failed login attempts are made during that time.

!!! warning
    If you have VS Code open with a Remote SSH session, it may attempt to reconnect automatically in the background, resetting the 90-minute timer and keeping your account locked. Close VS Code before waiting for the lock to clear.

To prevent this from happening again:

- Log into [ORCD OnDemand](https://orcd-ood.mit.edu) — this satisfies Duo authentication and removes the Duo requirement for SSH for a few hours.
- Use [SSH Control Channels](control-channels.md) to reduce how often you need to authenticate with Duo.

> - If you use VS Code, adjust the [VS Code Remote SSH settings](../recipes/vscode.md#adjust-the-remotessh-extension-settings).

Reviewer: ```suggestion
- If you use VS Code, adjust the [VS Code Remote SSH settings](../recipes/vscode.md#adjust-the-remotessh-extension-settings) to limit reconnection attempts.
```
